### PR TITLE
Add private key serialization.

### PR DESCRIPTION
**Pull request by chris-wood**
cc @rozbb

### CODE REVIEW

File: crypto.go [hunk 1]
```diff
@@ -432,8 +461,14 @@ func (s x25519Scheme) DeserializePrivate(enc []byte) (KEMPrivateKey, error) {
 		return nil, fmt.Errorf("Error deserializing X25519 private key")
 	}
 
+	copied := clampX25519(enc)
+	var zero [32]byte
+	if subtle.ConstantTimeCompare(copied, zero[:]) == 1 {
```
Comment (rozbb) [inline crypto.go:466]: This is insufficient to check for zero keys. You need to reduce modulo the curve order

File: crypto.go [hunk 2]
```diff
@@ -536,8 +583,14 @@ func (s x448Scheme) DeserializePrivate(enc []byte) (KEMPrivateKey, error) {
 		return nil, fmt.Errorf("Error deserializing X448 private key")
 	}
 
+	copied := clampX448(enc)
+	var zero [56]byte
+	if subtle.ConstantTimeCompare(copied, zero[:]) == 1 {
```
Comment (rozbb) [inline crypto.go:588]: Ditto

File: crypto.go (hunk 1 above)
Comment (chris-wood) [reply crypto.go:466]: Yeah, since it's a SHOULD, I removed that check.

**Comment by rozbb** [comment]
Also, it turns out that clamped scalars can never be 0 mod q. So clamping is sufficient and you don’t have to check anything

**Pull request closed without merging**